### PR TITLE
answered 2nd and 3rd questions

### PR DESCRIPTION
**Pull request by fuwai-so**

### CODE REVIEW

File: src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java [hunk 1]
```diff
@@ -51,20 +53,93 @@ public void test_length_basic() {
...
+                        maxStr = strContent;
+                    }
+                    if (minStr == null || minStr.length() > currentLength) {
+                        minStr = strContent;
+                    }
+                }
+            }
+        }
+
+        int different = maxStr.length() - minStr.length();
```
Comment (jflute) [inline src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java]: good variable name!

File: src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java [hunk 2]
```diff
@@ -51,20 +53,93 @@ public void test_length_basic() {
...
+        List<ColorBox> colorBoxList = new YourPrivateRoom().getColorBoxList();
+        String foo = null;
+        String bar = null;
+        for (ColorBox colorBox : colorBoxList) {
+            List<BoxSpace> spaceList = colorBox.getSpaceList();
+            for (BoxSpace space : spaceList) {
+                Object content = space.getContent();
+                if (content instanceof String) { // if the content is String type
+                    String strContent = (String) content;
+                    int currentLength = strContent.length();
```
Comment (jflute) [inline src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java]: The scope of the variable "currentLength" is wide.
The variable does not used by next logic "if (foo == null || bar == null) {".
(used by "if (foo.length() <= currentLength)..." statement)

So I think it should be moved to the immediately front of "if (foo.length() <= currentLength)...".
Generally a variable scope should be short for safety and readable.

File: src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java [hunk 3]
```diff
@@ -51,20 +53,93 @@ public void test_length_basic() {
...
+
+                    if (foo == null || bar == null) {
```
Comment (jflute) [inline src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java]: The logic works well. Thanks.
However other logics that are more simple may exist .
Can you reduce if-statement count?

File: src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java (hunk 2 above)
Comment (fuwai-so) [reply src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java]: I see! Thank you let me update it

File: src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java (hunk 3 above)
Comment (fuwai-so) [reply src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java]: Sure, let me try!

File: src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java (hunk 1 above)
Comment (fuwai-so) [reply src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java]: Thanks!

File: src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java [hunk 4]
```diff
@@ -177,13 +196,43 @@ public void test_length_findMaxColorSize() {
...
+        List<ColorBox> colorBoxList = new YourPrivateRoom().getColorBoxList();
+        String targetedColor = null;
+        for (ColorBox colorBox : colorBoxList) {
+            List<BoxSpace> spaceList = colorBox.getSpaceList();
+            for (BoxSpace space : spaceList) {
+                Object content = space.getContent();
+                if (content instanceof String) { // if the content is String type
+                    String strContent = (String) content;
+                    if (Pattern.matches("^Water.*", strContent)) {
+                        targetedColor = colorBox.getColor().getColorName();
```
Comment (jflute) [inline src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java]: If an another color-box that has "Water" content exists, the last color-box is selected now.
Comment about it for source-code reader.
(If no comment, reader may think "this logic is all-right?")

File: src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java [hunk 5]
```diff
@@ -167,6 +168,24 @@ public void test_length_calculateLengthSum() {
...
+        String strongestColor = null;
+        for (ColorBox colorBox : colorBoxList) {
+            List<BoxSpace> spaceList = colorBox.getSpaceList();
+            for (BoxSpace space : spaceList) {
+                Object content = space.getContent();
+                if (content instanceof String) { // if the content is String type
+                    String strContent = (String) content;
+                    int currentLength = strContent.length();
+                    if (maxStr == null || maxStr.length() < currentLength) {
+                        maxStr = strContent;
```
[the remark this answers is not in the document]
Comment (fuwai-so) [reply src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java]: Thanks! Let me update it

File: src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java [hunk 6]
```diff
@@ -177,13 +196,43 @@ public void test_length_findMaxColorSize() {
...
     public void test_startsWith_findFirstWord() {
+        List<ColorBox> colorBoxList = new YourPrivateRoom().getColorBoxList();
+        String targetedColor = null;
+        for (ColorBox colorBox : colorBoxList) {
+            List<BoxSpace> spaceList = colorBox.getSpaceList();
+            for (BoxSpace space : spaceList) {
+                Object content = space.getContent();
+                if (content instanceof String) { // if the content is String type
+                    String strContent = (String) content;
+                    if (Pattern.matches("^Water.*", strContent)) {
```
[the remark this answers is not in the document]
Comment (fuwai-so) [reply src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java]: Yes! That's true, I should use startsWith

File: src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java (hunk 4 above)
Comment (fuwai-so) [reply src/test/java/org/docksidestage/javatry/colorbox/Step11ClassicStringTest.java]: Sure, let me add the comment

**Comment by fuwai-so** [comment]
Sorry 久保さん
I just realized the pull request is pointing to your branch
let me close it and point back to my branch

**Pull request closed without merging**